superset-frontend/src/SqlLab/components/ResultSet.tsx

```diff
@@ -56,6 +55,21 @@ const EXPLORE_CHART_DEFAULT = {
   viz_type: 'table',
 };
 
+enum LIMITING_FACTOR {
+  QUERY = 'QUERY',
+  QUERY_AND_DROPDOWN = 'QUERY_AND_DROPDOWN',
+  DROPDOWN = 'DROPDOWN',
+  NOT_LIMITED = 'NOT_LIMITED',
+}
+
+function limitReached(displayLimitReached: boolean) {
+  return displayLimitReached;
+}
+
+function defaultDropdown(queryLimit: number, limitingFactor: string) {
+  return queryLimit === 1000 && limitingFactor === LIMITING_FACTOR.DROPDOWN;
```

Comment:
I don't remember, is the 1000 default query limit configurable? If so, we should read the configured value. Even if not, we should move this to a constant.